File: Britefury/DocPresent/Typesetting/VBox.py
##-*************************
##-* This program is free software; you can use it, redistribute it and/or modify it
##-* under the terms of the GNU General Public License version 2 as published by the
##-* Free Software Foundation. The full text of the GNU General Public License
##-* version 2 can be found in the file named 'COPYING' that accompanies this
##-* program. This source code is (C)copyright Geoffrey French 1999-2007.
##-*************************
from Britefury.DocPresent.Typesetting.Box import Box


class VBox (Box):
	def __init__(self, contents, glueType):
		w = reduce( lambda a, b: a + b.reqWidth(),  contents, 0.0 )
		h = reduce( lambda a, b: max( a, b.height() ),  contents, 0.0 )
		d = reduce( lambda a, b: max( a, b.depth() ),  contents, 0.0 )
		super( VBox, self ).__init__( w, h, d, contents, glueType )
	

File: java/BritefuryJ/DocPresent/Combinators/Primitive/VBox.java
//##* This program is free software; you can use it, redistribute it and/or modify it
//##* under the terms of the GNU General Public License version 2 as published by the
//##* Free Software Foundation. The full text of the GNU General Public License
//##* version 2 can be found in the file named 'COPYING' that accompanies this
//##* program. This source code is (C)copyright Geoffrey French 2008-2010.
//##************************
package BritefuryJ.DocPresent.Combinators.Primitive;

import java.util.List;

import BritefuryJ.DocPresent.DPElement;
import BritefuryJ.DocPresent.DPVBox;
import BritefuryJ.DocPresent.Combinators.PresentationContext;
import BritefuryJ.DocPresent.StyleSheet.StyleValues;

public class VBox extends AbstractBox
{
	private int refPointIndex;
	
	
	public VBox(Object children[])
	{
		super( children );
		this.refPointIndex = -1;
	}
	
	public VBox(Object children[], int refPointIndex)
	{
		super( children );
		this.refPointIndex = refPointIndex;
	}
	
	public VBox(List<Object> children)
	{
		super( children );
		this.refPointIndex = -1;
	}

	public VBox(List<Object> children, int refPointIndex)
	{
		super( children );
		this.refPointIndex = refPointIndex;
	}

	
	@Override
	public DPElement present(PresentationContext ctx, StyleValues style)
	{
		DPVBox element = new DPVBox( Primitive.vboxParams.get( style ) );
		element.setChildren( mapPresent( ctx, Primitive.useVBoxParams( style ), children ) );
		if ( refPointIndex != -1 )
		{
			element.setRefPointIndex( refPointIndex );
		}
		return element;
	}
}
